### DISCUSSION TITLE
Selecting inside/around HTML/React/XML elements

### DISCUSSION THREAD

**Discussion by gf3** [meow-edit/meow#244] [Q&A]
Is it possible to use <kbd>,</kbd> and <kbd>.</kbd> to select inside or around HTML-like elements?

E.g. with the following code:

```html
<div class="lol">
  Div
</div>

<p>
  Nice <span>work!</span>
</p>
```

And the cursor on `!`, using <kbd>,</kbd> would select `work!` and using <kbd>.</kbd> would select `<span>work!</span>`.

**Comment by snowiow** [comment]
Maybe this gets you started: 

```elisp
(meow-thing-register 'tags
    '(regexp "<.+>" "</.+>")
    '(regexp "<.+>" "</.+>"))
(add-to-list 'meow-char-thing-table '(?t . tags))
```

This works for your case and any inner tags. But for example when the cursor is somewhere on `Nice` it only highlights to the next closing tag, because this doesn't know what the matching opening tag is. 

I don't know if you can capture something in the first regex to use in the second one (that would solve the current problem). At least there is nothing mentioned in the docs.

**Reply by gf3** [reply]
thanks for the pointer @snowiow—maybe i'll see how evil-mode implements it

**Reply by snowiow** [reply]
There are also the `sgml-skip-tag-forward` and `sgml-skip-tag-backward` functions in sgml mode  which could be of help

**Reply by gf3** [reply]
@snowiow looks like i essentially want this without the evil → https://github.com/redguardtoo/evil-matchit

**Comment by DogLooksGood** [comment]
In fact you can register a function which receives no argument, returns two points in a cons.

**Reply by gf3** [reply]
@DogLooksGood i wonder if it might be worth integrating with `evil-matchit` for meow things, they just made evil an optional dependency: https://github.com/redguardtoo/evil-matchit/issues/137

**Reply by DogLooksGood** [reply]
Hmmm, Meow is zero dependency package.

If you are using web-mode, it provides all kinds of functions that you can use to find almost every positions. For example, `web-mode-element-beginning-position` and `web-mode-element-end-position`. Well, it depends on what major-mode you are using.
